src/analysis/retail/paladin/holy/modules/heroTalents/herald/Dawnlight.tsx

```diff
@@ -0,0 +1,210 @@
+import { formatNumber } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/paladin';
+import SpellIcon from 'interface/SpellIcon';
+import SpellLink from 'interface/SpellLink';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent, HealEvent } from 'parser/core/Events';
+import Combatants from 'parser/shared/modules/Combatants';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+
+class Dawnlight extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  protected combatants!: Combatants;
+
+  healingSource = new Map<
+    number,
+    {
+      totalAmount: number;
+      totalOverheal: number;
+      hits: { timestamp: number; amount: number; overheal: number }[];
+    }
+  >();
+
+  damageSource = new Map<number, { totalAmount: number; count: number }>();
+
+  refreshed = 0;
+
+  constructor(args: Options) {
+    super(args);
+    this.active = this.selectedCombatant.hasTalent(TALENTS.DAWNLIGHT_TALENT);
+
+    this.addEventListener(
+      Events.heal
+        .by(SELECTED_PLAYER)
+        .spell([
+          SPELLS.DAWNLIGHT_AOE_HEAL,
+          SPELLS.DAWNLIGHT_HEAL,
+          SPELLS.SUNS_AVATAR_HEAL,
+          SPELLS.SUNS_AVATAR_HEAL_SELF_APPLIED,
+        ]),
+      this.onHeal,
+    );
+
+    this.addEventListener(
+      Events.damage
+        .by(SELECTED_PLAYER)
+        .spell([
+          SPELLS.DAWNLIGHT_DAMAGE,
+          SPELLS.SUNS_AVATAR_DAMAGE,
+          SPELLS.SUNS_AVATAR_DAMAGE_SELF_APPLIED,
+        ]),
+      this.onDamage,
+    );
+  }
+
+  addToHealingSource(spellId: number, amount: number, overheal: number, timestamp: number) {
+    if (!this.healingSource.has(spellId)) {
+      this.healingSource.set(spellId, {
+        totalAmount: 0,
+        totalOverheal: 0,
+        hits: [],
+      });
+    }
+    const entry = this.healingSource.get(spellId)!;
+    entry.totalAmount += amount;
+    entry.totalOverheal += overheal;
+    entry.hits.push({ timestamp, amount, overheal });
+  }
+
+  addToDamageSource(spellId: number, amount: number) {
+    if (!this.damageSource.has(spellId)) {
+      this.damageSource.set(spellId, { totalAmount: 0, count: 0 });
+    }
+    const entry = this.damageSource.get(spellId)!;
+    entry.totalAmount += amount;
+    entry.count += 1;
+  }
+
+  get totalHealing() {
+    let sum = 0;
+    this.healingSource.forEach((entry) => (sum += entry.totalAmount));
+    return sum;
+  }
+
+  get totalOverhealing() {
+    let sum = 0;
+    this.healingSource.forEach((entry) => (sum += entry.totalOverheal));
+    return sum;
```

Comment:
You should use reduce instead for any summation logic. Apply where appropriate

```suggestion
    return this.healingSource.reduce((prev, cur) => prev + cur, 0)
```